### DISCUSSION TITLE
Break Infinity and Break Eternity in scratch

### DISCUSSION THREAD

**Discussion by Yoreni** [TurboWarp/extensions#1097] [Show and tell]
I made extensions that allows break infinity and break eternity to be used in scratch
Both Break Infinity and Break Extensions are javascript libraries made by Patashu 

Links to libraries
(Break Infinity)[https://github.com/Patashu/break_infinity.js]
(Break Eternity)[https://github.com/Patashu/break_eternity.js]

Both of these libraries allows you to have numbers bigger than 1.79e308 which is useful for incremental games.
 - break infinity allows up to ee15
 - break eternity allows up to 10↑↑1e308

The links to use the extensions are here
- break infinity: https://yoreni.github.io/scratch-extensions/breakInfinityScratch.js
- break eternity: https://yoreni.github.io/scratch-extensions/breakEternityScratch.js

**Comment by yuri-kiss** [comment]
which one is better, and does it support decimals?
________________________________
From: Yoreni ***@***.***>
Sent: Friday, October 13, 2023 5:33 AM
To: TurboWarp/extensions ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [TurboWarp/extensions] Break Infinity and Break Eternity in scratch (Discussion #1097)


I made extensions that allows break infinity and break eternity to be used in scratch
Both Break Infinity and Break Extensions are javascript libraries made by Patashu

Links to libraries
(Break Infinity)[https://github.com/Patashu/break_infinity.js]
(Break Eternity)[https://github.com/Patashu/break_eternity.js]

Both of these libraries allows you to have numbers bigger than 1.79e308 which is useful for incremental games.

  *   break infinity allows up to ee15
  *   break eternity allows up to 10↑↑1e308

The links to use the extensions are here

  *   break infinity: https://yoreni.github.io/scratch-extensions/breakInfinityScratch.js
  *   break eternity: https://yoreni.github.io/scratch-extensions/breakEternityScratch.js

**Reply by Yoreni** [reply]
They both support decimals, negitive numbers, extermly small decimals etc. 

Neither is better it depends which one is good for what you are doing. if you are not going to need numbers bigger than ee15 use break infinity otherwise break eternity

**Comment by yuri-kiss** [comment]
cool thanks :D!
________________________________

**Comment by yuri-kiss** [comment]
when im able to may i put this on my gallery?
________________________________

**Reply by Yoreni** [reply]
yes you may

**Comment by yuri-kiss** [comment]
Yay Thanks!
________________________________

**Comment by yuri-kiss** [comment]
i will put break eternity and have infinity hidden
________________________________

**Comment by NotAKrayer** [comment]
What about toPrecision?
